snap/snapcraft.yaml

```diff
@@ -0,0 +1,134 @@
+name: opensearch # you probably want to 'snapcraft register <name>'
+base: core20 # the base snap is the execution environment for this snap
+
+version: '2.2.0' # just for humans, typically '1.2+git' or '1.3.2'
+
+summary: 'OpenSearch: community-driven, Apache 2.0-licensed search and analytics suite.'
+description: |
+    OpenSearch is a community-driven, Apache 2.0-licensed open source search and 
+    analytics suite that makes it easy to ingest, search, visualize, and analyze data. 
+    Developers build with OpenSearch for use cases such as application search, 
+    log analytics, data observability, data ingestion, and more.
+
+grade: devel # must be 'stable' to release into candidate/stable channels
+
+confinement: strict # use 'strict' once you have the right plugs and slots
+
+
+assumes:
+  - command-chain
+  - snap-env
+  - snapd2.56
+
+
+system-usernames:
+  snap_daemon: shared
+
+
+plugs:
+  systemd-write:
+    interface: system-files
+    write:
+      - /etc/systemd/system/snap.opensearch.daemon.d
+
+hooks:
+  configure:
+    plugs:
+      - network
+      - network-bind
+      - systemd-write
+    environment:
+      OPS_ROOT: ${SNAP}/ops
+
+
+environment:
+  PROJECT_NAME: ${SNAPCRAFT_PROJECT_NAME}
+  PROJECT_VERSION: ${SNAPCRAFT_PROJECT_VERSION}
+
+  SNAP_LOG_DIR: ${SNAP_COMMON}/ops/snap/logs
+
+  OPENSEARCH_HOME: ${SNAP_DATA}
+  OPENSEARCH_JAVA_HOME: ${SNAP_DATA}/jdk
+  OPENSEARCH_PATH_CONF: ${SNAP_COMMON}/config
+  OPENSEARCH_TMPDIR: ${SNAP_COMMON}/tmp
+  OPENSEARCH_PLUGINS: ${SNAP_DATA}/plugins
+
+
+apps:
+  daemon:
+    daemon: simple
+    install-mode: disable
+    command: ops/start.sh
+    reload-command: ops/start.sh
+    plugs:
+      - network
+      - network-bind
+      - systemd-write
+      - mount-observe
+    environment:
+      OPS_ROOT: ${SNAP}/ops
+
+  setup:
+    command: ops/setup.sh
+    environment:
+      OPS_ROOT: ${SNAP}/ops
+      OPENSEARCH_PATH_CERTS: ${OPENSEARCH_PATH_CONF}/certificates
+
+  plugin-add:
+    command: ops/plugins/add.sh
+
+  plugins-list:
+    command: ops/plugins/list.sh
+
+  plugins-remove:
+    command: ops/plugins/remove.sh
+
+parts:
+  python-requirements:
+    plugin: python
+    python-packages:
+      - ruamel.yaml
+
+  wrapper-scripts:
+    plugin: nil
+    source: ./scripts
+    source-type: local
+    override-build: |
+      target_dir="${SNAPCRAFT_PART_INSTALL}/ops"
+      
+      rm -rf "${target_dir}"
+      
+      mkdir -p "${target_dir}"/logs/
```

Comment:
I am not sure why this is created in a read only location that would not be writable from the snap, more so considering `SNAP_LOG_DIR: ${SNAP_COMMON}/ops/snap/logs`